game-core/src/main/java/games/strategy/triplea/ui/mapdata/MapData.java

```diff
@@ -811,15 +779,18 @@ private Rectangle getBoundingRectWithTranslate(
   }
 
   public Optional<Image> getTerritoryEffectImage(final String effectName) {
-    // TODO: what does this cache buy us? should we still keep it?
-    if (effectImages.get(effectName) != null) {
-      return Optional.of(effectImages.get(effectName));
-    }
-    Optional<Image> effectImage = loadImage("territoryEffects/" + effectName + "_large.png");
-    if (effectImage.isEmpty()) {
-      effectImage = loadImage("territoryEffects/" + effectName + ".png");
+    try (ResourceLoader loader = ResourceLoader.getMapResourceLoader(mapNameDir)) {
```

Comment:
This method looks up territory effects at runtime. It's not clear if we could load them all in construction very easily in case a map would add territory effects as some sort of trigger. Hence we need to instantiate a resource loader here to support dynamic lookup of territory effects. Not ideal, but a trade-off..